### PR TITLE
XML documentation for FastTree binary classification.

### PR DESCRIPTION
**Pull request by codemzs**
towards #2522

### CODE REVIEW

File: docs/api-reference/algo-details-fasttree.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+regression problems, boosting builds a series of such trees in a step-wise
+fashion and then selects the optimal tree using an arbitrary differentiable loss
+function.
+
+MART learns an ensemble of regression trees, which is a decision tree with
+scalar values in its leaves. A decision (or regression) tree is a binary
+tree-like flow chart, where at each interior node one decides which of the two
+child nodes to continue to based on one of the feature values from the input. At
+each leaf node, a value is returned. In the interior nodes, the decision is
+based on the test 'x &lt;= v' where x is the value of the feature in the input
```
Comment (shmoradims) [inline docs/api-reference/algo-details-fasttree.md]: >x &lt;= v [](start = 19, length = 9)

x <= v

xml escaping is not required in markdown

File: docs/api-reference/algo-details-fasttree.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+can be produced by a regression tree are all the piece-wise constant functions.
+          
+The ensemble of trees is produced by computing, in each step, a regression tree
+that approximates the gradient of the loss function, and adding it to the
+previous tree with coefficients that minimize the loss of the new tree. The
+output of the ensemble produced by MART on a given instance is the sum of the
+tree outputs.
+
+* In case of a binary classification problem, the output is converted to a
+  probability by using some form of calibration.>
```
Comment (shmoradims) [inline docs/api-reference/algo-details-fasttree.md]: >> [](start = 48, length = 1)

remove

File: docs/api-reference/algo-details-fasttree.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* In case of a regression problem, the output is the predicted value of the
+  function.
+* In case of a ranking problem, the instances are ordered by the output value of
+  the ensemble.
+
+For more information see:
+
+[Wikipedia: Gradient boosting (Gradient tree
```
Comment (shmoradims) [inline docs/api-reference/algo-details-fasttree.md]: add bullet: *

File: src/Microsoft.ML.FastTree/FastTreeArguments.cs [hunk 4]
```diff
@@ -55,14 +55,16 @@ public enum EarlyStoppingRankingMetric
...
+        /// [FastTree(Options)](xref:"Microsoft.ML.TreeExtensions.FastTree(Microsoft.ML.BinaryClassificationCatalog.BinaryClassificationTrainers,Microsoft.ML.Trainers.FastTree.FastTreeBinaryTrainer.Options)).
         /// </summary>
         [TlcModule.Component(Name = LoadNameValue, FriendlyName = UserNameValue, Desc = Summary)]
         public sealed class Options : BoostedTreeOptions, IFastTreeTrainerFactory
         {
 
             /// <summary>
-            /// Whether to use derivatives optimized for unbalanced training data.
+            /// Options for the <see cref="FastTreeBinaryTrainer"/> as used in
+            /// [FastTree(Options)](xref:"Microsoft.ML.TreeExtensions.FastTree(Microsoft.ML.BinaryClassificationCatalog.BinaryClassificationTrainers,Microsoft.ML.Trainers.FastTree.FastTreeBinaryTrainer.Options)).
```
Comment (shmoradims) [inline src/Microsoft.ML.FastTree/FastTreeArguments.cs]: revert this change

Comment (codemzs) [reply src/Microsoft.ML.FastTree/FastTreeArguments.cs]: Why?

Comment (codemzs) [reply src/Microsoft.ML.FastTree/FastTreeArguments.cs]: oh wait crap

File: src/Microsoft.ML.FastTree/FastTreeClassification.cs [hunk 5]
```diff
@@ -100,7 +100,28 @@ private static IPredictorProducing<float> Create(IHostEnvironment env, ModelLoad
     /// <summary>
     /// The <see cref="IEstimator{TTransformer}"/> for training a decision tree binary classification model using FastTree.
     /// </summary>
-    /// <include file='doc.xml' path='doc/members/member[@name="FastTree_remarks"]/*' />
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// To create this trainer, use [FastTree](xref:Microsoft.ML.TreeExtensions.FastTree(Microsoft.ML.BinaryClassificationCatalog.BinaryClassificationTrainers,System.String,System.String,System.String,System.Int32,System.Int32,System.Int32,System.Double))
+    /// or [FastTree(Options)](xref:"Microsoft.ML.TreeExtensions.FastTree(Microsoft.ML.BinaryClassificationCatalog.BinaryClassificationTrainers,Microsoft.ML.Trainers.FastTree.FastTreeBinaryTrainer.Options)).
+    ///
+    /// [!include[io](~/../docs/samples/docs/api-reference/io-columns-binary-classification.md)]
```
Comment (natke) [inline src/Microsoft.ML.FastTree/FastTreeClassification.cs]: Is that the right path?

File: src/Microsoft.ML.FastTree/FastTreeClassification.cs [hunk 6]
```diff
@@ -100,7 +100,28 @@ private static IPredictorProducing<float> Create(IHostEnvironment env, ModelLoad
...
+    ///
+    /// ### Trainer Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Machine learning task | Binary classification |
+    /// | Is normalization required? | No |
+    /// | Is caching required? | No |
+    /// | Required NuGet in addition to Microsoft.ML | Microsoft.ML.FastTree |
+    ///
+    /// [!include[algorithm](~/../docs/samples/docs/api-reference/algo-details-fasttree.md)]
```
Comment (natke) [inline src/Microsoft.ML.FastTree/FastTreeClassification.cs]: Right path?

Comment (codemzs) [reply src/Microsoft.ML.FastTree/FastTreeClassification.cs]: I followed the path used in AP

Comment (codemzs) [reply src/Microsoft.ML.FastTree/FastTreeClassification.cs]: I also asked @shmoradims the same question

File: src/Microsoft.ML.FastTree/FastTreeClassification.cs (hunk 5 above)
Comment (codemzs) [reply src/Microsoft.ML.FastTree/FastTreeClassification.cs]: Yes, based on the path used in AP
